presto-hive/src/main/java/com/facebook/presto/hive/PrestoS3FileSystem.java

```diff
@@ -326,7 +326,9 @@ public FSDataOutputStream create(Path path, FsPermission permission, boolean ove
             throw new IOException("File already exists:" + path);
         }
 
-        createDirectories(stagingDirectory.toPath());
+        if (!stagingDirectory.exists()) {
+            createDirectories(stagingDirectory.toPath());
+        }
```

Comment:
Yes, this does seem to work. That is, stagingDirectory.isDirectory() seems to return true if the staging directory is a symlink to a directory.

Do you think it's worth adding another test (for the sake of increasing code coverage) that tests that an appropriate error is thrown if the staging directory exists but is actually a file instead of a directory? And what about yet another test that verifies that an appropriate error is returned if the staging directory happens to be a symlink to a file rather than a directory?